src/Core/Injector/Injector.php

```diff
@@ -876,8 +909,10 @@ public function unregisterAllObjects()
      */
     public function get($name, $asSingleton = true, $constructorArgs = null)
     {
+        $object = null;
```

Comment:
Copied from slack:

abstract it to
 - find service spec (either by hasService() or locateConfigFor()
 - construct found service

Turn

```if (A) {
 construct(A)
}
elseif (B) {
  construct(B)
}
```

into

```spec = A || B
construct(spec)
```

you know that Injector::get() (current implementation) calls ->instantiate() five times right?